### ISSUE TITLE
consume long time for predicting validation output

### ISSUE THREAD

**Issue by A7medRamadan2018** [lvapeab/nmt-keras#120]
I train my model with 63k training samples for 15 minutes but when valid on validation set (7000 samples) it take over hour !!

**Comment by lvapeab** [comment]
It seems that something is wrong. Can you give me more information? Are you using GPU? Can you share your config file?

**Comment by A7medRamadan2018** [comment]
yes I use GPU on google colaboratory, I am using the default configuration: https://github.com/lvapeab/nmt-keras/blob/master/config.py  but  
**MAX_INPUT_TEXT_LEN = 300, 
Use_Cuda= True**
these are different parameters 
if I make validation data = 100 sample, this makes it fast. 
but my valid is 7000 too long time

**Comment by lvapeab** [comment]
Indeed, the validation time is linearly proportional w.r.t. the number of samples (currently, no batch decoding is implemented). So you may want to validate on a smaller set (but big enough to take generalizable takeaways). This time is also proportional to the number of max steps allowed in the beam search (`MAX_OUTPUT_TEXT_LEN_TEST`). If the model hasn't learned yet to generate sentences of the adequate length, it will take longer, as the search will make this number of steps.

You should also take a look to your data and check whether it makes sense setting `MAX_INPUT_TEXT_LEN=300`. If you are doing regular word-level MT, this value seems too high.
